### PR TITLE
Import Netsuite employees to fyle

### PR DESCRIPTION
**Pull request by harshithaputtaswamy**

### CODE REVIEW

File: src/app/core/models/updated-configuration.ts [hunk 1]
```diff
@@ -12,5 +12,10 @@ export type UpdatedConfiguration = {
     oldValue: string,
     newValue: string
   },
-  autoCreateDestinationEntity: boolean
+  autoCreateDestinationEntity: boolean,
+  importNetsuiteEployee?: {
```
Comment (NileshPant1999) [inline src/app/core/models/updated-configuration.ts:16]: spelling

Comment (NileshPant1999) [reply src/app/core/models/updated-configuration.ts:16]: Employee 😛

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.ts [hunk 2]
```diff
@@ -471,6 +488,16 @@ export class ConfigurationComponent implements OnInit {
     }
   }
 
+  showImportEmployeeOprion(employeeMappedTo) {
```
Comment (NileshPant1999) [inline src/app/netsuite/settings/netsuite-configurations/configuration/configuration.component.ts:491]: spelling

File: src/app/core/models/updated-configuration.ts [hunk 3]
```diff
@@ -12,5 +12,10 @@ export type UpdatedConfiguration = {
     oldValue: string,
     newValue: string
   },
-  autoCreateDestinationEntity: boolean
+  autoCreateDestinationEntity: boolean,
+  importNetsuiteEmployee?: {
```
Comment (NileshPant1999) [inline src/app/core/models/updated-configuration.ts:16]: why is this field optional

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.scss [hunk 4]
```diff
@@ -42,7 +42,6 @@
     cursor: default;
     color: #5C98E5;
     font-size: 19px !important;
-    margin-bottom: 18px !important;
```
Comment (NileshPant1999) [inline src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.scss:45]: why did we removed this.

**Review by NileshPant1999 — APPROVED** [review]

File: src/app/core/models/updated-configuration.ts (hunk 3 above)
Comment (harshithaputtaswamy) [reply src/app/core/models/updated-configuration.ts:16]: updating this field only when the import employee toggle is enabled

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.scss (hunk 4 above)
Comment (harshithaputtaswamy) [reply src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.scss:45]: This style was applied for info-icon, in html <space-between center> was includes, removed <center> here and the hardcoded margin in css, the alignment is fine now

File: src/app/core/models/general-setting.model.ts [hunk 5]
```diff
@@ -20,4 +20,5 @@ export type GeneralSetting = {
     updated_at?: Date;
     workspace?: number;
     import_vendors_as_merchants?: boolean;
+    import_netsuite_employees?: boolean;
```
Comment (ashwin1111) [inline src/app/core/models/general-setting.model.ts:23]: why is this optional?

File: src/app/core/models/updated-configuration.ts [hunk 6]
```diff
@@ -12,5 +12,10 @@ export type UpdatedConfiguration = {
...
+    oldValue: boolean,
+    newValue: boolean
+  },
+  showMappingsChange: boolean
```
Comment (ashwin1111) [inline src/app/core/models/updated-configuration.ts:20]: we already have importNetsuiteEmployee right, can't we make use of that?

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.html [hunk 7]
```diff
@@ -88,6 +80,33 @@ <h2 mat-dialog-title>
...
+        <div fxFlex="4">
+          <mat-icon [inline]="true" class="configurations-dialog--info-icon">info</mat-icon>
+        </div>
+        <div fxFlex="96">
+          <b>Importing Employee data</b> from Netsuite will <b>overwrite the data</b> stored in Fyle.
+        </div>
+      </div>
+      <div fxLayout="row" fxLayoutAlign="space-between center">
+        <div fxFlex="96" [style.margin-top]="'20px'">
+          Employee data like Department, Supervisor, E-mail ID, etc. will be imported from Netsuite and updated in Fyle. <a href="https://netsuite.fyleapps.tech/workspaces">Learn More</a>
```
Comment (ashwin1111) [inline src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.html:106]: please add a TODO to replace help article link

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.ts [hunk 8]
```diff
@@ -35,6 +37,7 @@ export class ConfigurationDialogComponent implements OnInit {
 
   setup() {
     const that = this;
+    that.showMappingsChange = that.updatedConfiguration.showMappingsChange;
```
Comment (ashwin1111) [inline src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.ts:40]: same

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.ts [hunk 9]
```diff
@@ -45,6 +48,9 @@ export class ConfigurationDialogComponent implements OnInit {
       exportType = exportType.charAt(0).toUpperCase() + exportType.substr(1).toLowerCase();
       that.additionalWarning = `${exportType} would require an Expense account for successful export. You can import this by enabling the toggle below or creating a manual map from the Category Mapping section.`;
     }
+    if (that.updatedConfiguration.importNetsuiteEmployee.newValue && !that.updatedConfiguration.importNetsuiteEmployee.oldValue) {
```
Comment (ashwin1111) [inline src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.ts:51]: that.updatedConfiguration.importNetsuiteEmployee && <remaining>

File: src/app/core/models/general-setting.model.ts (hunk 5 above)
Comment (harshithaputtaswamy) [reply src/app/core/models/general-setting.model.ts:23]: Already on-boarded customer might not want to use this feature

File: src/app/core/models/updated-configuration.ts (hunk 6 above)
Comment (harshithaputtaswamy) [reply src/app/core/models/updated-configuration.ts:20]: In the html component of configuaration-dialogue, by default the summary of changes related to export mappings change is shown, but when the only change is enabling the feature toggle then we don't want to show the heading related to export mappings change

File: src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.ts (hunk 8 above)
Comment (harshithaputtaswamy) [reply src/app/netsuite/settings/netsuite-configurations/configuration/configuration-dialog/configuration-dialog.component.ts:40]: addressed above

File: src/app/core/models/general-setting.model.ts (hunk 5 above)
Comment (ashwin1111) [reply src/app/core/models/general-setting.model.ts:23]: no no, this key should be always present in the payload no?

File: src/app/core/models/updated-configuration.ts (hunk 6 above)
Comment (ashwin1111) [reply src/app/core/models/updated-configuration.ts:20]: you can check that with the value present in importNetsuiteEmployee key no? you will be adding value for this key (importNetsuiteEmployee) only when the toggle gets enabled/disabled no?

File: src/app/core/models/general-setting.model.ts (hunk 5 above)
Comment (harshithaputtaswamy) [reply src/app/core/models/general-setting.model.ts:23]: Right, if it is not selected or not visible then we are sending False

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**